tests/test_cases/test_iteration_vhdl/test_iteration.py

```diff
@@ -30,9 +30,7 @@
 from cocotb.result import TestFailure
 
 
-# This test crashes Riviera-PRO 2019.10 (at least); skip to avoid hanging the
-# tests. See issue #1854 for details.
-@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith("riviera") and cocotb.SIM_VERSION.startswith("2019.10"))
+@cocotb.test()
```

Comment:
Take my comment as an action. The result of this test should be justified as to why they are different than the other number in this test for Riviera; then that number should be added to the test. As long as the number "makes sense", the test is really there just to prevent regression. Since we have a version of Riviera in our CI it important we actually run this test.

Perhaps we need a function to determine what the expected count is for the given simulator and version, which can double as a skip condition?

Reply:
I moved the expected count to a function and added the justifications for Active-HDL differences.

I don't have access to Modelsim so I can't provide a justification for why it has a higher object count.

Reply:
I see you decided to keep the default expected count. I think we should be skipping the test unless we are sure we have a valid count. The function could return `0` if there is no confirmed count and you could use that in the test decorator.

```suggestion
@cocotb.test(skip=(total_object_count() == 0))
```

What do you think?